### PR TITLE
Add share function

### LINKED ISSUE #78: [Feature Request] Add a share button
here

### PR DESCRIPTION
**Pull request by Icarus-Xu**
Use system share intent to send music file uri to other app that recognize it.

Tested on Telegram, Wechat & QQ.

Closes #78.

### CODE REVIEW

File: app/src/main/java/io/github/zyrouge/symphony/ui/components/SongCard.kt [hunk 1]
```diff
@@ -1,5 +1,8 @@
 package io.github.zyrouge.symphony.ui.components
 
+import android.content.Intent
+import android.util.Log
```
Comment (zyrouge) [inline app/src/main/java/io/github/zyrouge/symphony/ui/components/SongCard.kt:4]: `Log` doesn't seem to be used anywhere. Kindly remove it.

File: app/src/main/java/io/github/zyrouge/symphony/ui/components/SongCard.kt [hunk 2]
```diff
@@ -274,6 +277,29 @@ fun SongDropdownMenu(
...
+                try {
+                    context.activity.startActivity(Intent(Intent.ACTION_SEND).apply {
+                        addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION)
+                        putExtra(Intent.EXTRA_STREAM, song.uri)
+                        type = context.activity.contentResolver.getType(song.uri)
+                    })
+                } catch (e: Exception) {
+                    Toast.makeText(
+                        context.activity,
+                        "${context.symphony.t.ShareFailed} ${e.localizedMessage}",
```
Comment (zyrouge) [inline app/src/main/java/io/github/zyrouge/symphony/ui/components/SongCard.kt:297]: Make it `Share failed (Error: <error message>)`.

**Review by zyrouge — CHANGES_REQUESTED** [review]

**Comment by Icarus-Xu** [comment]
done, sorry for the delay

**Comment by Icarus-Xu** [comment]
TY

**Pull request merged**